scanpy/experimental/pp/_highly_variable_genes.py

```diff
@@ -0,0 +1,341 @@
+            means=means,
+            variances=variances,
+            residual_variances=np.mean(residual_gene_vars, axis=0).astype(
+                np.float32, copy=False
+            ),
+            highly_variable_rank=medianrank_residual_var,
+            highly_variable_nbatches=highly_variable_nbatches.astype(np.int64),
+            highly_variable_intersection=highly_variable_intersection,
+        )
+    )
+    df = df.set_index(adata.var_names)
+
+    # Sort genes by how often they selected as hvg within each batch and
+    # break ties with median rank of residual variance across batches
+    df.sort_values(
+        ['highly_variable_nbatches', 'highly_variable_rank'],
+        ascending=[False, True],
+        na_position='last',
+        inplace=True,
+    )
+
+    high_var = np.zeros(df.shape[0])
+    high_var[:n_top_genes] = True
+    df['highly_variable'] = high_var.astype(bool)
+    df = df.loc[adata.var_names, :]
+
+    if inplace:
+        adata.uns['hvg'] = {'flavor': 'pearson_residuals', 'computed_on': computed_on}
+        logg.hint(
+            'added\n'
+            '    \'highly_variable\', boolean vector (adata.var)\n'
+            '    \'highly_variable_rank\', float vector (adata.var)\n'
+            '    \'highly_variable_nbatches\', int vector (adata.var)\n'
+            '    \'highly_variable_intersection\', boolean vector (adata.var)\n'
+            '    \'means\', float vector (adata.var)\n'
+            '    \'variances\', float vector (adata.var)\n'
+            '    \'residual_variances\', float vector (adata.var)'
+        )
+        adata.var['means'] = df['means'].values
+        adata.var['variances'] = df['variances'].values
+        adata.var['residual_variances'] = df['residual_variances']
+        adata.var['highly_variable_rank'] = df['highly_variable_rank'].values
+        if batch_key is not None:
+            adata.var['highly_variable_nbatches'] = df[
+                'highly_variable_nbatches'
+            ].values
+            adata.var['highly_variable_intersection'] = df[
+                'highly_variable_intersection'
+            ].values
+        adata.var['highly_variable'] = df['highly_variable'].values
+
+        if subset:
+            adata._inplace_subset_var(df['highly_variable'].values)
+
+    else:
+        if batch_key is None:
+            df = df.drop(
+                ['highly_variable_nbatches', 'highly_variable_intersection'], axis=1
+            )
+        if subset:
+            df = df.iloc[df.highly_variable.values, :]
+
+        return df
+
+
+def highly_variable_genes(
+    adata: AnnData,
+    theta: float = 100,
+    clip: Optional[float] = None,
+    n_top_genes: Optional[int] = None,
+    batch_key: Optional[str] = None,
+    chunksize: int = 1000,
+    flavor: Literal['pearson_residuals'] = 'pearson_residuals',
+    check_values: bool = True,
+    layer: Optional[str] = None,
+    subset: bool = False,
+    inplace: bool = True,
+) -> Optional[pd.DataFrame]:
```

Comment:
```suggestion
def highly_variable_genes(
    adata: AnnData,
    *,
    theta: float = 100,
    clip: Optional[float] = None,
    n_top_genes: Optional[int] = None,
    batch_key: Optional[str] = None,
    chunksize: int = 1000,
    flavor: Literal['pearson_residuals'] = 'pearson_residuals',
    check_values: bool = True,
    layer: Optional[str] = None,
    subset: bool = False,
    inplace: bool = True,
) -> Optional[pd.DataFrame]:
```

Would it make sense for this to not allow positional arguments, since the second argument differs from the other function?

Reply:
I agree, can add them all as keyword.

Reply:
I think I'm not familiar with the syntax here - does the `*` mean that all arguments after that need to be supplied with explicit keyword? What is the other function you are referring to?
@giovp do you want to make the commit (also the one in the comment raised below) or should I?

Reply:
done